specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json

```diff
@@ -0,0 +1,1275 @@
+          "maxLength": 300
+        },
+        "questions": {
+          "type": "array",
+          "description": "List of questions associated with the answer.",
+          "maxLength": 100,
+          "minLength": 1,
+          "items": {
+            "type": "string"
+          }
+        },
+        "metadata": {
+          "type": "array",
+          "description": "List of metadata associated with the answer.",
+          "maxLength": 10,
+          "items": {
+            "$ref": "#/definitions/MetadataDTO"
+          }
+        }
+      }
+    },
+    "MetadataDTO": {
+      "type": "object",
+      "description": "Name - value pair of metadata.",
+      "additionalProperties": false,
+      "required": [
+        "name",
+        "value"
+      ],
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Metadata name.",
+          "maxLength": 100,
+          "minLength": 1
+        },
+        "value": {
+          "type": "string",
+          "description": "Metadata value.",
+          "maxLength": 500,
+          "minLength": 1
+        }
+      }
+    },
+    "ErrorResponse": {
+      "type": "object",
+      "description": "Error response. As per Microsoft One API guidelines - https://github.com/Microsoft/api-guidelines/blob/vNext/Guidelines.md#7102-error-condition-responses.",
+      "additionalProperties": false,
+      "properties": {
+        "error": {
+          "description": "The error object.",
+          "allOf": [
+            {
+              "$ref": "#/definitions/Error"
+            }
+          ]
+        }
+      }
+    },
+    "Error": {
+      "type": "object",
+      "description": "The error object. As per Microsoft One API guidelines - https://github.com/Microsoft/api-guidelines/blob/vNext/Guidelines.md#7102-error-condition-responses.",
+      "additionalProperties": false,
+      "required": [
+        "code"
+      ],
+      "properties": {
+        "code": {
+          "description": "One of a server-defined set of error codes.",
+          "$ref": "#/definitions/ErrorCode"
+        },
+        "message": {
+          "type": "string",
+          "description": "A human-readable representation of the error."
+        },
+        "target": {
+          "type": "string",
+          "description": "The target of the error."
+        },
+        "details": {
+          "type": "array",
+          "description": "An array of details about specific errors that led to this reported error.",
+          "items": {
+            "$ref": "#/definitions/Error"
+          }
+        },
+        "innerError": {
+          "description": "An object containing more specific information than the current object about the error.",
+          "$ref": "#/definitions/InnerErrorModel"
+        }
+      }
+    },
+    "ErrorCode": {
+      "type": "string",
+      "description": "Human readable error code.",
+      "x-enumNames": [
```

Comment:
See if https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-enum would be suitable in this case